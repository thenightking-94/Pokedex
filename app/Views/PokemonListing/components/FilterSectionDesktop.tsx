import {  Checkbox, FormControl, Grid, InputLabel, ListItemText, MenuItem, OutlinedInput } from '@mui/material';
import Select from '@mui/material/Select';
import React, { ReactElement } from 'react';
  

export default function FilterSectionDesktop (props): ReactElement{
    const {handleChangeGender, handleChangeType, pokemonGenderDataFiltered, pokemonTypeDataFiltered, PokemonTypes, PokemonGender} = props
   
    return(
        <div className='filter-section'>
            <Grid className='grid-offset' container spacing={2} alignItems='center' justifyContent={'space-between'}>
                <Grid item sm={4}>
                    <div className='type-section'>
                        <p className='label'>Type</p>
                        <FormControl className='type-checkbox'>
                            <InputLabel id="checkbox-label-type"><span className='desc'>{'Normal'}</span><span className='bold'>{' +5 More'}</span></InputLabel>
                            <Select
                            labelId="checkbox-label-type"
                            id="multiple-checkbox-type"
                            multiple
                            value= {pokemonTypeDataFiltered}
                            onChange= {handleChangeType}
                            input={<OutlinedInput label="Types" />}
                            renderValue={(selected) => selected.join(', ')}
                            >
                            {PokemonTypes?.map((i) => (
                                <MenuItem className='menu-item' key={i} value={i}>
                                    <Checkbox checked={pokemonTypeDataFiltered.indexOf(i) > -1} />
                                    <ListItemText primary={i} />
                                </MenuItem>
                            ))}
                            </Select>
                        </FormControl>
                    </div>
                </Grid>
              
                <Grid item sm={4}>
                    <div className='gender-section'>
                        <p className='label'>Gender</p>
                        <FormControl className='gender-checkbox'>
                            <InputLabel id="checkbox-label-gender"><span className='desc'>{'Male'}</span><span className='bold'>{' +2 More'}</span></InputLabel>
                            <Select
                            labelId="checkbox-label-gender"
                            id="multiple-checkbox-gender"
                            multiple
                            value= {pokemonGenderDataFiltered}
                            onChange= {handleChangeGender}
                            input={<OutlinedInput label="Gender" />}
                            renderValue={(selected) => selected.join(', ')}
                            >
                            {PokemonGender?.map((i) => (
                                <MenuItem className='menu-item' key={i} value={i}>
                                <Checkbox checked={pokemonGenderDataFiltered.indexOf(i) > -1} />
                                <ListItemText primary={i} />
                                </MenuItem>
                            ))}
                            </Select>
                        </FormControl>
                    </div>
                </Grid>
               
                <Grid item sm={4}>
                    <div className='stats-section'>
                        <p className='label'>Stats</p>
                        <FormControl className='stats-box'>
                            <InputLabel id="checkbox-label"><span className='desc'>{'HP'}</span><span className='bold'>{' +5 More'}</span></InputLabel>
                                <Select
                                labelId="checkbox-label"
                                id="multiple-checkbox"
                                multiple
                                value= {undefined}
                                onChange={undefined}
                                input={<OutlinedInput label="Stats" />}
                                >
                                </Select>
                        </FormControl>
                    </div>
                </Grid>
              
            </Grid>
        </div>
    )

}