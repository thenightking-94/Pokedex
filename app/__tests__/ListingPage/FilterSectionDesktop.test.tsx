import { render , waitFor, screen,} from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import FilterSectionDesktop from '../../Views/PokemonListing/components/FilterSectionDesktop'

const queryClient = new QueryClient()

test("simplified-version", async () => {
    act(() => {
      render(<FilterSectionDesktop/>);
    });

    await waitFor(() => {
        screen.findByText("Stats");
        expect(screen.getAllByText("Stats")).toBeTruthy();
      });
});

describe("Jest Snapshot testing suite", () => {
    it("Matches DOM Snapshot", () => {
      const domTree = renderer.create( 
      <QueryClientProvider client={queryClient}>
        <FilterSectionDesktop/>
     </QueryClientProvider>).toJSON();
      expect(domTree).toMatchSnapshot();
    });
  });