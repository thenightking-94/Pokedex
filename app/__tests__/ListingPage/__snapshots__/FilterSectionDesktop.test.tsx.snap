// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  className="filter-section"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 grid-offset css-6ejd34-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 css-kgs8gf-MuiGrid-root"
    >
      <div
        className="type-section"
      >
        <p
          className="label"
        >
          Type
        </p>
        <div
          className="MuiFormControl-root type-checkbox css-1nrlq1o-MuiFormControl-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            id="checkbox-label-type"
          >
            <span
              className="desc"
            >
              Normal
            </span>
            <span
              className="bold"
            >
               +5 More
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="checkbox-label-type multiple-checkbox-type"
              className="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="multiple-checkbox-type"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Types
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 css-kgs8gf-MuiGrid-root"
    >
      <div
        className="gender-section"
      >
        <p
          className="label"
        >
          Gender
        </p>
        <div
          className="MuiFormControl-root gender-checkbox css-1nrlq1o-MuiFormControl-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            id="checkbox-label-gender"
          >
            <span
              className="desc"
            >
              Male
            </span>
            <span
              className="bold"
            >
               +2 More
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="checkbox-label-gender multiple-checkbox-gender"
              className="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="multiple-checkbox-gender"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Gender
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 css-kgs8gf-MuiGrid-root"
    >
      <div
        className="stats-section"
      >
        <p
          className="label"
        >
          Stats
        </p>
        <div
          className="MuiFormControl-root stats-box css-1nrlq1o-MuiFormControl-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            id="checkbox-label"
          >
            <span
              className="desc"
            >
              HP
            </span>
            <span
              className="bold"
            >
               +5 More
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="checkbox-label multiple-checkbox"
              className="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="multiple-checkbox"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Stats
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
